### PR TITLE
Add api stuff

### PR DESCRIPTION
**Pull request by b4l**
Since #811 is in a broken state we might merge this first to get up and running. Also this contains a better history for the rust project as well as improvements to the project not present in #811 . There is still some parts we have to drag in from the latter but can wait for a follow up PR.

### CODE REVIEW

File: api/Cargo.toml [hunk 1]
```diff
@@ -1,10 +1,23 @@
 [package]
-name = "api"
+name = "bedrock"
```
[the remark this answers is not in the document]
Comment (b4l) [reply api/Cargo.toml:2]: If would prefer to save the namespace to not end up using `api::server::api`

[the remark this answers is not in the document]
Comment (gberaudo) [reply api/Cargo.toml:2]: Do you prefer we use `backend`?

[the remark this answers is not in the document]
Comment (b4l) [reply api/Cargo.toml:2]: Maybe we should use the `ngm` prefix like `ngm-ui`, `ngm-backend`, `ngm-tools`, `ngm-....` this wold make a clean entrypoint

[the remark this answers is not in the document]
Comment (gberaudo) [reply api/Cargo.toml:2]: We can decide tomorrow at the daily what is the best naming.
The problem I see with `viewer` is that we don't know if we speak about the frontend or backend part,

**Pull request closed without merging**